### ISSUE TITLE
Advance setting and power consumption

### ISSUE THREAD

**Issue by PatrikMihalik** [Piotr-Grochowski/eu.blebox#4]
Hello, 

It is great app but some possibility is missing.
It is posssibility from another your device blebox- tempsensor. 
In advance setting is posssible use consumption when is on and off and statusnindicator is possible to set on celsius. This is not possible to set on thermobox. 
Will be posssible to add this function? 

Thank you

![Screenshot_2023-04-04-18-16-49-562_app homey](https://user-images.githubusercontent.com/129721129/229858424-11cf13be-e009-444a-b4d7-486ec25f5b1e.jpg)
![received_1365760810881393](https://user-images.githubusercontent.com/129721129/229858772-66e1a301-87b5-45a6-a3b4-2818ea5394b5.jpeg)

**Comment by Piotr-Grochowski** [comment]
Hello,
Regarding the power consumption: 
1. It's NOT a bug in thermoBox -> the energy consumption is constant for this device, so the functionality in Homey is OK - please look at the specification on BleBox page: https://blebox.eu/en/product/thermobox/
2. The other screen you've pasted (with IP 192.168.0.29) is not from tempSensor, in tempSensor there is also constant power usage <1W, there is no username and password setting and there are more informational fields like product name, API level, etc. Please check that again - what device is this? Is this for sure a BleBox device from recent app version?
3. Regarding the status indicator - first thing it's not set in the advanced settings. It's in Settings->Status indicator. For thermostat devices it's not possible on HOMEY to set the status indicator - it's blocked by the Homey app. I have another thermostat in my office - from ecobee - and I can see in the Homey App that Settings->Status indicator is "Not configurable". So it's not the app problem, but it's a limitation of Homey app itself.
4. I can see your recent comment under my app on Homey AppStore: "not possible switch on/off and change range of temperature like in original app" - please tell me what you mean. The UI component used in my app is standard Homey component for all thermostats. But maybe I can add some functionality (if BleBox API supports it and there is a possibility to add it to Homey) - please tell me more about what you mean. Please note, that I'm not a BleBox employee, but just a community developer. I don't have a heating installation with thermoBox installed, so I can't check everything - I use help and testing of such users as You to add/debug/test additional functionalities.

**Comment by PatrikMihalik** [comment]
Thank you for your powerfull explanation. I understand that you are not blebox developer. I try to clearly describe what I mean. I need switch on and switch off the relay which is inblebox when is reached some temperature. This is not possible in homey. Is possible set "when" is temperature lower/higher but in "then" is not possible switch on or switch of relay.this switch function is very important. Second thing is range of temperature in homey. In standard app from blebox is possible change range from 0 to 110'C. Because sensor have -55 to 125 'C.This two thing is for me necessary to use it with hot water tank heater and photovoltaic Solax. Need switch on water tank heater to 75'C if the baterry is full and FVE production is over 2,5kW.Tha last thing is possible have power consumption, if you can add power ussage when on and when off. it will be great for me.I just set the heater consumption when is on and homey calculate the power usage automatically. icon is just for visualization on mainboard, is not neccesary. I can help you with testing new version if you want to help me and other who buy this thermosensor. Thank you for your time. Patrikodesláno z mobiluDne 4. 4. 2023 22:40 napsal uživatel Piotr-Grochowski ***@***.***>:
Hello,
Regarding the power consumption:
It's NOT a bug in thermoBox -> the energy consumption is constant for this device, so the functionality in Homey is OK - please look at the specification on BleBox page: https://blebox.eu/en/product/thermobox/The other screen you've pasted (with IP 192.168.0.29) is not from tempSensor, in tempSensor there is also constant power usage <1W, there is no username and password setting and there are more informational fields like product name, API level, etc. Please check that again - what device is this? Is this for sure a BleBox device from recent app version?Regarding the status indicator - first thing it's not set in the advanced settings. It's in Settings->Status indicator. For thermostat devices it's not possible on HOMEY to set the status indicator - it's blocked by the Homey app. I have another thermostat in my office - from ecobee - and I can see in the Homey App that Settings->Status indicator is "Not configurable". So it's not the app problem, but it's a limitation of Homey app itself.I can see your recent comment under my app on Homey AppStore: "not possible switch on/off and change range of temperature like in original app" - please tell me what you mean. The UI component used in my app is standard Homey component for all thermostats. But maybe I can add some functionality (if BleBox API supports it and there is a possibility to add it to Homey) - please tell me more about what you mean. Please note, that I'm not a BleBox employee, but just a community developer. I don't have a heating installation with thermoBox installed, so I can't check everything - I use help and testing of such users as You to add/debug/test additional functionalities.

**Comment by Piotr-Grochowski** [comment]
I've had a long conversation with BleBox employee, who is my internal support for developing this integration with Homey. It looks like I've covered all the cases that were mentioned in BleBox thermoBox manual - room heating/cooling, but not the case you have - with hot water tank. There are several things in Homey BleBox app that I have to redesign from scratch. I will do it, but it will take me some time. Please be patient - I should be able to do everything you need, including wider temperature range, additional flow action cards, and estimated power consumption when on/off. I hope you will be then able to give me more than just one * in the app review ;)

Happy Easter in the meantime!

**Comment by PatrikMihalik** [comment]
Thank you very much for great news. 
I think it will be great if we can use it with full potential.
Adds card to flow is necessary. 🙂

To the icon, I notice that it shows set up temperature, if its set higher from actual is red and its below its blue. It s OK. Great will see booth temperature, set and actual. 
![IMG_20230406_162434](https://user-images.githubusercontent.com/129721129/230419805-25be6505-7498-4089-b2bf-674277db34cd.jpg)

If you need my support fot testing, it is no problem and help you. I wanted to send this device back and order tempbox and switch box separately. But is better have this in one device if you upgrade integration. 
How long it takes for you appproximately? 

Thank you again.

**Comment by Piotr-Grochowski** [comment]
I've had some time today, so I've made the changes and they are ready for your test!

1. Please delete thermoBox device from Homey
2. Please install test app from here: https://homey.app/pl-pl/app/eu.blebox/BleBox/test/
3. Please add thermoBox device to Homey and make a test.

A few things to know:
1. Standard Homey UI for thermostats supports only max 40 Celsius Degrees, so I had to remove this UI. Instead of this there is a slider to control the temperature and "Target temperature" sensor on the sensors tab. It's Homey's limitation, not mine, so it can't be done in another way :( 
2. There are some trigger flow cards you can use:
- when target temperature changes
- when safety temperature goes in some ranges
- when measured temperature changes
3. There are some action flow cards you can use:
- change the state of the thermostat (OFF, ON (2 modes depending on your config), BOOST)
- change the target temperature

I hope it will meet all your need and cover all your scenarios. Please let me know if you have any problems and also please let me know if it works OK for you!

**Comment by PatrikMihalik** [comment]
Hello, 

I look on it and install guickly. 
But now I'm not at home for better testing. 
For first sign it a pitty that is no possibility to have fully thermostat circle from 5-80 degree. The slider is not so good choice I think. There is no other way how set temperature? Maybe  label wit manual set temperature or something like that?

2. When card is OK,but in then card is just set thermostat state to off not to on and there is change the target temperature. 
3. Power consumption when on/off still missing.
4. Temperature setting icon disappear.

This is my guick overview, tomorrow test it better.

Thank you for quick first upgrade.

**Comment by Piotr-Grochowski** [comment]
Hello,
Thanks for quick review. Here are some answers:
Ad. 1 I also think that it's a real shame that Homey limits thermostat UI usage to 40 Celsius degrees, which makes it usable only for room thermostat conditions. I'll try to ask Athom if they can change it in the future, but I don't think there are much chances. Your case is a real niche at the moment I think. The slider is the only option at the moment - unfortunately there is no UI component in Homey to get user input... - please look at this webpage for Homey developers - here are all UIs that can be used for different capabilities (paragraph "UI Components"): https://apps.developer.homey.app/the-basics/devices/capabilities But on the other hand - in your case will you really use manual temperature change? I think you'll set one or send those via flows (I don't know you exact rules), not set them manually...
Ad. 2 I tested the flows - I could add Then card "Set thermostat state to..." and then using selector I could select: Off, On (temperature set), On (no temperature set), Boost. In my case due to thermoBox configuration the option On (no temperature set) did nothing - I think it's when no probes are connected to thermoBox (I used two - main and safety). Please check it once again and if you can't change Off to another option via selector then please send me some screenshots. If you can, please use not only mobile app but also my.homey.app webpage on your PC.
Ad. 3 To be honest I didn't test that, but I can see in the configuration that it should be there - in Settings -> Advanced Settings -> Power usage when off and Power usage when on - initially both values set to 1, but you should be able to change them. If you don't have those options please send me a screenshot. Unfortunately during Easter I won't be able to test thermoBox, because I left it in my office by mistake...
Ad. 4 You mean the little "XX oC" on the icon of the thermoBox on the "Devices" tab? It was available for thermostat UI by standard (with the changing color behavior). For other devices it's different. I think you should be able to go to your device's Settings -> Status Indicator and there change what you want to see (measured temperature, safety temperature or target temperature). I didn't test it either. 

Please let me know, I may be able to change something and provide a new test version during Easter.  

Piotr

**Comment by PatrikMihalik** [comment]
Hello, you are so fast. 😉it is great. 

To termostat slider. i understand, but from homey is not happy solution. In some cases is neccesary to set more like 40 degree, like water tank acumullation, radiator heating, bojler and so on.if we want set room temperature 40 is to enough. If homey can make this range adjustable it will be great solution. And as you wrote is not neccesary to set temperature manualy. 

To 2. Thus button was hide under the off buton, it was guiet tricky for me 😀. It is OK. 

To 3. To power usage, this button still missing. i add screenshot froom homey. 
![IMG_20230408_121038](https://user-images.githubusercontent.com/129721129/230716809-ee078439-503c-47cc-94d1-bf1f4213d574.jpg)

Thank you for your great response and work. 


To 4. Is not possible to change the status indicator to anything. See picture below
![Screenshot_2023-04-08-12-10-12-129_app homey](https://user-images.githubusercontent.com/129721129/230716864-6efa9190-9dcf-4c16-8da1-4e34a9bfa2c6.jpg)

**Comment by Piotr-Grochowski** [comment]
Hello,
Ad. 1 So let's leave it like this for now. Maybe I'll make a second version (in one or two separate drivers) for people who use thermoBox as a standard room thermostat - with standard Homey UI. It would be great if Athom changes the UI to support more than 40degC.

Ad. 2 OK, great
Ad. 3 I deep dived into this topic and I think I know why is that so - that's because of the class of the device (thermostat, not socket/wallplug/light/etc.). The energy calculation won't be calculated right by Homey, that's why it's only constant value for the controller only (even though I made two separate fields in thermoBox configuration - when ON and when OFF). Why is that so? Please think about this scenario:
a) we configure thermoBox to use e.g. 1W when Off and 1500W when On.
b) the water in the tank is 25 deg C, thermoBox is Off
c) we set thermoBox to 75 deg C and turn it On (with temperature set)
d) thermoBox sends the signal to the water heater and it starts to use 1500W and Homey calculates so
e) the water in the tank is being heated and in e.g. 30 minutes it reaches 75deg C
d) thermoBox sends a signal to water heater to stop heating and it uses 1W. BUT thermoBox is still ON and Homey calculates it as 1500W even through it's not true!
e) so if you would leave your thermoBox to control the temperature of water tank to 75 deg C for 24h/7d it would be ON 24/7 and Homey would calculate 1500W 24/7 even if the heater works only part of this time.

What do you think about this? Maybe I'm wrong?
Ad. 4 I checked that but it looks like app developer has no impact on which capability can be used in Status Indicator. I think for thermostat device class it's blocked and they show the values from standard Homey thermostat UI. But as I don't use it in this driver, but the slider - nothing shows up in the status indicator. So I won't be able to change that...

So to summarize - I think nothing can be done now to make all this better due to Homey limitations (especially thermostat UI - as a source of all the problems)... The most important is if you can use this driver as you need it - I mean if you can make all the flows you need, etc? Or is some scenario undoable and I should make something for you?

**Comment by Piotr-Grochowski** [comment]
Hello again,
I published a new test version 3.9.2 of the app - two thermoBox drivers are available - for standard room thermostat scenario and high temperature scenario (like yours). I've seen in other water tank heaters drivers for homey that they also use slider for this... 
I had to make some changes due to this splitting into two separate drivers - please check if the flow trigger and action cards do still exist (especially to turn off/on/boost the thermoBox) and let me know if you have any problems with this version.

**Comment by PatrikMihalik** [comment]
OK i test this new version and send my suggestions. To the icon.I try to find some solution like you and one of it is use virtual device, which show bojler temperature and I see it directly. The best one will be see both temperature in one window but... 🤣 I dont know how does it do.But for now is OK. See picture below, the same virtual device use fot see solax battery.

![IMG_20230408_154808](https://user-images.githubusercontent.com/129721129/230725094-da02b43f-b2e0-4a3c-be4f-da2ede4013c0.jpg)

To the power consumption problematic:

I investigate how it works in BLEBOX application. 

ON/OFF works like switch. But just in ON position I can control 24/7 the setting temperature. It's not socket which have 2 status on/off and calculate consumption when switch on. 

Instead of that in blebox original, when I have switch ON, there is information topení,/nečinné and this is what we need to calculate consumption.
When is topení calculate f.e. 1500W or other sett. value and when is nečinné not calculate. Looks below
![Screenshot_2023-04-08-16-04-00-040_eu blebox wBox~2](https://user-images.githubusercontent.com/129721129/230725574-083545a8-2c56-4dd4-85ea-b35244f38be5.jpg)

![Screenshot_2023-04-08-16-03-38-530_eu blebox wBox~2](https://user-images.githubusercontent.com/129721129/230725580-e8598445-d009-4cce-a1e3-66f18cf8a911.jpg)

**Comment by Piotr-Grochowski** [comment]
Could you please translate me into english this Topeni/necinne?

**Comment by PatrikMihalik** [comment]
It is native from blebox in czech. 

But. i think it is topení=heating and nečinné=not working (stop working or stand-by) hard to describe.
But is the like in this window I think. Try to look in Homey. How change the status when change temperature below and overr the water tank temperature.
![IMG_20230408_163233](https://user-images.githubusercontent.com/129721129/230726912-1195c81b-2092-4233-81e3-35bdea8cb8bd.jpg)

**Comment by PatrikMihalik** [comment]
Today i try to change language in Blebox to english for translate this two words. 

Topení=heating it is deppend on mode (heating/cooling) 

Nečinné= Idle (termostat is on mode but not heat) 

I hope that this information help you to set calculating the power consumption. 🙂

**Comment by Piotr-Grochowski** [comment]
OK, I found in the API where it is reported. I'll try to use this info, we'll see if power consumption approximation will work properly. I'll need a few days for this.

**Comment by PatrikMihalik** [comment]
To 1.

![Screenshot_2023-04-11-15-20-24-480_eu blebox wBox](https://user-images.githubusercontent.com/129721129/231175865-f3762005-87e2-4b64-b981-25f2f9da2dc8.jpg)

ON and heating

{"thermo":{"state": 1,"operatingState": 1,"desiredTemp": 4000,"mode": 1,"minimumTemp": 500,"maximumTemp": 8000,"safety": {"eventReason": 0,"triggered": []},"safetyTempSensor": {"sensorId": null }},"sensors":[{"id": 0,"type": "temperature","value": 3868,"state": 2}]}

**Comment by PatrikMihalik** [comment]
OFF
![Screenshot_2023-04-11-15-24-36-405_eu blebox wBox](https://user-images.githubusercontent.com/129721129/231177158-819f7756-62d2-4cf7-be66-464fe70edf12.jpg)

{"thermo":{"state": 0,"operatingState": 0,"desiredTemp": 4000,"mode": 1,"minimumTemp": 500,"maximumTemp": 8000,"safety": {"eventReason": 0,"triggered": []},"safetyTempSensor": {"sensorId": null }},"sensors":[{"id": 0,"type": "temperature","value": 4068,"state": 2}]}

**Comment by PatrikMihalik** [comment]
And last status

Is ON but in IDLE (temperature reached) 
![Screenshot_2023-04-11-15-26-16-571_eu blebox wBox](https://user-images.githubusercontent.com/129721129/231177604-f5c53718-9ffe-47c9-a628-33997c8e753c.jpg)

{"thermo":{"state": 1,"operatingState": 0,"desiredTemp": 4000,"mode": 1,"minimumTemp": 500,"maximumTemp": 8000,"safety": {"eventReason": 0,"triggered": []},"safetyTempSensor": {"sensorId": null }},"sensors":[{"id": 0,"type": "temperature","value": 4106,"state": 2}]}

**Comment by Piotr-Grochowski** [comment]
Thanks, everything looks OK, so it's weird. Please send me this diagnostic report - maybe there I will find the answer why is that not working.

**Comment by PatrikMihalik** [comment]
I am sending report

6cbc3af5-e143-45c5-be0d-5cecb330aef6

**Comment by PatrikMihalik** [comment]
I mean the "OperatingState" is the value which we searching for.  That change idle and heating.

**Comment by PatrikMihalik** [comment]
Anything new in this case Piotr? Can I help in other way?

**Comment by Piotr-Grochowski** [comment]
Hello,
I've just published a new TEST version of my app. I couldn't test it with the thermoBox through, I hope it will work as I predicted.
First - delete your thermoBox from Homey and after updating the app to 3.11.0 please add thermoBox again. It should have a new power measurement sensor (approximated). In thermoBox settings you'll find 3 new inputs: power usage when Idle, On and Boost - please fill in with your warm water heater power usage (please note than thermoBox itself consumes around 1W when Idle). The power usage sensor should calculate the power usage. Due to some thermoBox lags, preferred pooling should be set to 5000ms (default value) = 5s. So the value will be calculated with 5s resolution.

**Comment by Piotr-Grochowski** [comment]
Hello, I'm happy you've got all the functionalities you need. I'll be happy if you could correct now your review of my app on Homey AppStore with a new rating >1* :)

Regarding statistics - power calculation is made by homey and it's not stored on thermoBox, so the driver will not show totals and averages. It's a common approach of all the drivers (the drivers show statistics only if the device itself calculates them). Other Homey Apps should be used for totals and statistics - like Power By The Hour.

**Comment by Piotr-Grochowski** [comment]
Hello, due to some changes in Homey, it looks like we can now use standard thermostatUI instead of slider. I've made a new testversion there: https://homey.app/pl-pl/app/eu.blebox/BleBox/test/  - please remove your thermoBox and after installing this test version add it again (but with thermoBox device, not thermoBox (high temperature)). Please tell me if it now works OK for you with thermostatUI.

**Comment by PatrikMihalik** [comment]
Hello Piotr,this is great news.I have just one question. I have a lot of flows connect to old version. Do you think,that will be possible just upgrade this test version high temperature? Because when I delete device and add new one I lost conection to flow and to calculating kwh per month in another app.

**Comment by PatrikMihalik** [comment]
Hello again, I install new version. In mobile homey shows setting temperature,great upgrade.
![Screenshot_2023-09-17-21-18-30-978_app homey-edit](https://github.com/Piotr-Grochowski/eu.blebox/assets/129721129/d4e55645-6213-4fc5-87e9-3d35709542b3)

And termostat works great and looks great too.
![Screenshot_2023-09-17-21-18-04-090_app homey-edit](https://github.com/Piotr-Grochowski/eu.blebox/assets/129721129/7de5a55b-4a13-4146-b910-0cd15208e115)

Problem is just missing power consumption like in your previous test version when is termostat in heating mode. And last problem is in setting temperature on then card. There is range and it is not possible to set exact number.

**Comment by Piotr-Grochowski** [comment]
Hello,
Regarding power usage – I will add this from the version made specially for you to this standard version of the driver. Give me a couple of days.

Regarding setting temperature from a flow – it’s a standard Homey UI, I can set exact temperature:

[Obraz zawierający tekst, zrzut ekranu, Czcionka, linia  Opis wygenerowany automatycznie]

Or maybe you’ve meant something different?

Pozdrawiam,
Piotr Grochowski
www.vmp.com.pl<http://www.vmp.com.pl/> | ***@***.******@***.***> | tel. +48-501-510-524

VMP Group sp. z o.o. | ul.  Zawiła 65E | 30-390 Kraków
NIP: 6762580246 | KRS: 0000839593


Od: PatrikMihalik ***@***.***>
Odpowiedź do: "Piotr-Grochowski/eu.blebox" ***@***.***>
Data: niedziela, 17 września 2023 o 21:34
Do: "Piotr-Grochowski/eu.blebox" ***@***.***>
DW: Piotr Grochowski ***@***.***>, State change ***@***.***>
Temat: Re: [Piotr-Grochowski/eu.blebox] Advance setting and power consumption (Issue #4)


Hello again, I install new version. In mobile homey shows setting temperature,great upgrade.
[Screenshot_2023-09-17-21-18-30-978_app homey-edit]<https://user-images.githubusercontent.com/129721129/268524221-d4e55645-6213-4fc5-87e9-3d35709542b3.jpg>

And termostat works great and looks great too.
[Screenshot_2023-09-17-21-18-04-090_app homey-edit]<https://user-images.githubusercontent.com/129721129/268524229-7de5a55b-4a13-4146-b910-0cd15208e115.jpg>

Problem is just missing power consumption like in your previous test version when is termostat in heating mode. And last problem is in setting temperature on then card. There is range and it is not possible to set exact number.

**Comment by PatrikMihalik** [comment]
OK,thank you very much Piotr. I will wait for power usage version.

Regarding to the set the temperature,when I set it through web app it works good but is not possible to write exact number,just move with the driver and trying to set correct temperature.

If I want to set the temperature via mobile app  with every move driver jump from minimum to maximum temperature, sometime stopped on random value. I try to add video later.

**Comment by PatrikMihalik** [comment]
I am sending short video and screen how it works on solax,there is possible set number directly. There is no range.

![Screenshot_2023-09-18-11-38-49-460_app homey-edit](https://github.com/Piotr-Grochowski/eu.blebox/assets/129721129/e1f938cc-ff19-4281-8999-dedc9cf1d632)

And here is video with driver and trying to set 70 degree,it is impossible via mobile.


https://github.com/Piotr-Grochowski/eu.blebox/assets/129721129/ec7f1a39-c3b7-42a0-9e0b-6c3e1dc6b789

**Comment by Piotr-Grochowski** [comment]
This is regardless of me. This is how Athom created the UI - for WHEN card you can write down the exact number (for thermoBox too), but when setting the thermostat to a temperature - you have to use this slider. For me it works exactly the same on the web app and on the phone (I've got iPhone). If it works not right on you Android phone - make a support ticket to Athom - bacause how sliders work - it's on them with the whole Homey app. Maybe they have a bug in Android version.

**Comment by PatrikMihalik** [comment]
Ok Piotr,thank you for your testing. I try to solve it with homey and I will wait for your power consumption upgrade. 😉 have a nice evening and bug thanks.
